File: Python/programmers/etc/level2/멀리_뛰기.py
MOD = 1234567

def solution(n):
    dp = [0]*2001
    dp[1], dp[2] = 1, 2
    for i in range(3, n+1):
        dp[i] = (dp[i-1]+dp[i-2])%MOD
    return dp[n]


n = 4
print(solution(n)) # 5

n = 3
print(solution(n)) # 3

File: Java/programmers/etc/level2/멀리뛰기.java
package CodeTest.Java.programmers.etc.level2;

public class 멀리뛰기 {
    public int solution(int n) {
        int MOD = 1234567;
        int[] dp = new int[2001];
        dp[1] = 1; dp[2] = 2;
        
        for (int i=3; i<=n; i++) {
            dp[i] = (dp[i-1]+dp[i-2])%MOD;
        }
        
        return dp[n];
    }   
}
